sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -280,6 +280,9 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case r @ ShowTableProperties(rt: ResolvedTable, propertyKey) =>
       ShowTablePropertiesExec(r.output, rt.table, propertyKey) :: Nil
 
+    case AnalyzeTable(_: ResolvedTable, _, _) | AnalyzeColumn(_: ResolvedTable, _, _) =>
+      throw new AnalysisException("ANALYZE TABLE is not supported for v2 tables.")
```

Comment:
shall we fail it earlier in analyzer rules?

Reply:
Do you want a separate rule to handle these? There will be more commands not supported in v2.

Btw, we have similar checks in this file:
```scala
    case DescribeColumn(_: ResolvedTable, _, _) =>
      throw new AnalysisException("Describing columns is not supported for v2 tables.")
```

Reply:
ah ok, then it's fine.